ui/src/app/utils/machines.ts

```diff
@@ -45,5 +51,92 @@ export const allMachineTypes = fp.map(({ price, preemptiblePrice, ...details })
 // See https://broadworkbench.atlassian.net/browse/SATURN-1337
 export const validLeonardoMachineTypes = allMachineTypes.filter(({memory}) => memory >= 4);
 
+export const findMachineByName = (machineToFind: string) => fp.find(({name}) => name === machineToFind, allMachineTypes);
+
 export const diskPrice = 0.04 / 730; // per GB hour, from https://cloud.google.com/compute/pricing
 export const dataprocCpuPrice = 0.01; // dataproc costs $0.01 per cpu per hour
+
+export const machineStoragePrice = ({masterDiskSize, numberOfWorkers, workerDiskSize }) => {
+  if (numberOfWorkers && workerDiskSize) {
+    return (masterDiskSize + numberOfWorkers * workerDiskSize) * diskPrice;
+  } else {
+    return masterDiskSize * diskPrice;
+  }
+};
+
+export const machineStorageCostBreakdown = ({masterDiskSize, numberOfWorkers, workerDiskSize}) => {
+  const costs = [];
+  if (numberOfWorkers && workerDiskSize) {
+    costs.push(`${formatUsd(masterDiskSize * diskPrice)}/hr Master Disk`);
+    costs.push(`${formatUsd((numberOfWorkers * workerDiskSize) * diskPrice)}/hr Worker Disks`);
+  } else {
+    costs.push(`${formatUsd(masterDiskSize * diskPrice)}/hr Disk`);
+  }
+  return costs;
+};
+
+export const machineRunningPrice = ({
+  computeType,
+  masterDiskSize,
+  masterMachineName,
+  numberOfWorkers = 0,
+  numberOfPreemptibleWorkers = 0,
+  workerDiskSize,
+  workerMachineName
+}) => {
+  const masterMachine = findMachineByName(masterMachineName);
+  const workerMachine = workerMachineName && findMachineByName(workerMachineName);
+
+  if (computeType === ComputeType.Dataproc && !workerMachine) {
+    return NaN;
+  }
+
+  const dataprocPrice = computeType === ComputeType.Dataproc
+    ? fp.sum([
+      (masterMachine.cpu + ((numberOfWorkers + numberOfPreemptibleWorkers) * workerMachine.cpu)) * dataprocCpuPrice,
+      numberOfWorkers * workerMachine.price,
+      numberOfPreemptibleWorkers * workerMachine.preemptiblePrice,
+    ])
+    : 0;
+  return fp.sum([
+    dataprocPrice,
+    masterMachine.price,
+    machineStoragePrice({masterDiskSize: masterDiskSize, numberOfWorkers: numberOfWorkers, workerDiskSize: workerDiskSize})
```

Comment:
number of preemptible workers is not considered in this calculation - is that intentional?

Reply:
It is included in the dataproc price. This could be named clearer.

Reply:
The dataproc price just includes machine prices. I'm asking why preemptible workers are not considered on this specific line for the storage price computation.

Reply:
because Terra doesn't consider them: https://github.com/DataBiosphere/terra-ui/blob/cf5ec4408db3bd1fcdbcc5302da62d42e4d03ca3/src/components/ClusterManager.js#L85

Reply:
although, looking at GCE docs, I think it should.